### DISCUSSION TITLE
Can delete from base collection but not from auth collection

### DISCUSSION THREAD

**Discussion by philmayes** [pocketbase/pocketbase#1639] [Q&A] [answered]
Hi, like the title says, the second delete fails:
```
try {
  await pb.collection('userdata').delete(id);
  print('deleted userdata for $id');
  await pb.collection('users').delete(id);
  print('deleted user     for $id');
} on ClientException catch(e) {
  if (kDebugMode) {
    print('_deleteRemote exception: $e');
    print('the error: status code ${e.statusCode} ${e.response['message']}');
    return false;
  }
}
```
For the second delete, I get response: {code: 404, message: The requested resource wasn't found., data: {}}, originalError: null}
I have updated the server to v0.11.3 and Flutter client to 0.7.0
The debug info is:
```
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = '_pb_users_auth_' OR LOWER([[name]])='_pb_users_auth_' LIMIT 1
[0.00ms] SELECT `users`.* FROM `users` WHERE `users`.`id`='30tvqajvxo3d9ml' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'userdata' OR LOWER([[name]])='userdata' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = '1nlq4xi6gwtbf89' OR LOWER([[name]])='1nlq4xi6gwtbf89' LIMIT 1
[0.00ms] SELECT `userdata`.* FROM `userdata` WHERE (`userdata`.`id`='30tvqajvxo3d9ml') AND (COALESCE('30tvqajvxo3d9ml', '') != COALESCE('', '')) LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections`
[0.00ms] DELETE FROM `userdata` WHERE `id`='30tvqajvxo3d9ml'
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'users' OR LOWER([[name]])='users' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = '_pb_users_auth_' OR LOWER([[name]])='_pb_users_auth_' LIMIT 1
[0.00ms] SELECT `users`.* FROM `users` WHERE (`users`.`id`='30tvqajvxo3d9ml') AND (COALESCE([[users.id]], '') = COALESCE(NULL, '')) LIMIT 1
2023/01/18 18:04:52 sql: no rows in result set
```
Thanks for any advice.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not able to reproduce the issue locally.

Make sure that you are passing the correct user id (in the above example it seems that the same id is used for both `userdata` and `users`).

If that's not the case, then check whether your request client satisfy the `users` "Delete API rule".

**Comment by philmayes** [comment]
Thanks for replying so promptly, and sorry for not doing the same :-)
What appears to be happening is that the first delete cancels authorization. Before the 1st delete:
```
print(pb.authStore.isValid);
print(pb.authStore.model.id);
print('local id $id');
```
produces
```
true
m6vror5ywnx4xfr
local id m6vror5ywnx4xfr
```
Before the 2nd delete, it produces:
```
false
_deleteRemoteUser unknown exception: NoSuchMethodError: The getter 'id' was called on null.
Receiver: null
```
When I interpolate 
await pb.collection('users').authWithPassword(username, password);
between the two deletes, the second one works.
My assumption was that authorization remains valid until pb.authStore.clear(); is called (or until the Auth record authentication token expires in 1209600 seconds.) Thanks for any advice -- Phil

And thank you so much for this project!

**Reply by ganigeorgiev** [reply]
> What appears to be happening is that the first delete cancels authorization. 

Hm, yes and no.

I think the main issue is that you are passing the **same id for both `userdata` and `users` records and this will not work** (_unless you've manually set both collections records to have the same id, which is rarely the case_).

You are right though that the `pb.authStore` will be automatically cleared after deleting the "logged in" auth record. Similarly to `update()`, when calling `delete()` we are updating the `pb.authStore` if the updated/deleted record id matches with the one from the store (this is currently documented only in the [jsdoc](https://github.com/pocketbase/js-sdk/blob/b123521190adccdfa7c294dad7ecc3d48abe29b3/src/services/RecordService.ts#L207-L225) and [dartdoc](https://github.com/pocketbase/dart-sdk/blob/master/lib/src/services/record_service.dart#L122-L149) of the method).

There is one issue with this and it is that if you are using custom ids it could end up matching a record from different collection. I'll provide a fix sometime later today in the SDKs to check also whether the updated/deleted record collection name matches with the one in `pb.authStore.model` before updating/clearing the store.

**Reply by ganigeorgiev** [reply]
The JS and Dart SDKs were updated to check the collection name in addition to the record id on auth record update/delete:

- https://github.com/pocketbase/js-sdk/releases/tag/v0.10.1
- https://pub.dev/packages/pocketbase/changelog

_But once again, based on the initial provided code snippet and the sql logs, if you are not using custom ids, then your issue is most likely caused by an incorrect `userdata`/`users` id (you are using the same `30tvqajvxo3d9ml` id for both requests which is almost never the case if the ids are autogenerated)._

**Reply by philmayes** [reply]
Thank you so much! I don't have a database background, and it never occurred to me that this could be a problem.
